### PR TITLE
Escape single quotes writing response files

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #10623

### CODE REVIEW

File: tests/runner.py [hunk 1]
```diff
@@ -922,6 +922,8 @@ def expect_fail(self, cmd, **args):
     Return the stderr of the subprocess.
     """
     proc = run_process(cmd, check=False, stderr=PIPE, **args)
+    if proc.returncode == 0:
```
Comment (kripken) [inline tests/runner.py]: is this for debugging? then shouldn't it be `!=`? Or maybe I don't understand the motivation here.

Comment (sbc100) [reply tests/runner.py]: yes this is for debugging failures.

This is `expect_fail` so we expect that return code to be non-zero.   If its *is* zero then we don't want to sliently swallow the stderr (like we are currently doing).

Comment (sbc100) [reply tests/runner.py]: Hmm. I can think of a nicer way maybe..

**Comment by kripken** [comment]
The test here failed on the windows roller, https://logs.chromium.org/logs/emscripten-releases/buildbucket/cr-buildbucket.appspot.com/8886836011363004144/+/steps/Emscripten_testsuite__upstream__other_/0/stdout

**Comment by sbc100** [comment]
See https://github.com/emscripten-core/emscripten/pull/10635

**Pull request merged**